x/programs/test/programs/simple_token/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use wasmlanche_sdk::Context;
+use wasmlanche_sdk::{public, state_keys, types::Address, Program};
+
+const INITIAL_SUPPLY: i64 = 123456789;
+
+/// The program state keys.
+#[state_keys]
+pub enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// The balance of the token by address. Key prefix 0x3 + address.
+    Balance(Address),
+
+    Allowance([Address; 2]),
+
+    Owner,
+}
+
+pub fn get_owner(program: &Program<StateKey>) -> Address {
+    program
+        .state()
+        .get(StateKey::Owner)
+        .expect("failed to get total supply")
+}
+
+pub fn owner_check(program: &Program<StateKey>, actor: Address) {
+    assert!(
+        get_owner(program) == actor,
+        "caller is required to be owner"
+    )
+}
```

Comment:
You could actually write an extension trait for the `Context` to call `context.is_owner()` directly.